### ISSUE TITLE
regression with v4.30.6

### ISSUE THREAD

**Issue by jerome-laforge** [mikefarah/yq#1501]
With v4.30.6, we have this error: `Error: Cannot multiply !!map with !!null`

we don't have it with  v4.30.5.
I don't have all yaml fragment files to merge (there are a complex workflow in our CI), but I give you usecase when I find the yaml files that generate this error.

**Comment by Wohlraj** [comment]
Was going to write about this, I noticed it only happens when merge files, where the first of these files is only containing carriage returns and at least 3 of them:
```bash
$ ls
2newlines.yaml  3newlines.yaml  notempty.yaml
$ cat 2newlines.yaml 


$ cat 3newlines.yaml 



$ cat notempty.yaml 
test:
$ yq ea '. as $item ireduce ({}; . * $item )' 2newlines.yaml notempty.yaml 
test:
$ yq ea '. as $item ireduce ({}; . * $item )' 3newlines.yaml notempty.yaml 
Error: Cannot multiply !!map with !!null
```
This was not an issue with 4.30.5.

**Comment by mikefarah** [comment]
Got it, will fix in the next release 👍🏼

**Comment by jerome-laforge** [comment]
I have just tested with complied yq from the source code.
Your fix works like a cham.

Thank for your support

**Comment by mikefarah** [comment]
Fixed in 4.30.7
